File: src/main/java/com/tk/stock/service/StockService.java
package com.tk.stock.service;

import com.tk.stock.model.PageBean;
import com.tk.stock.model.Stock;

import java.util.List;
import java.util.Map;

public interface StockService {
    void initAllStockData();
    PageBean findAll(PageBean pageBean);
    void addStocks(List<Stock> stockList);
    PageBean search(PageBean pageBean,Map map);
}

File: src/main/resources/static/js/stockService.js
app.service("stockService", function ($http) {
    this.findAll = function (pageNum) {
        return $http.get('api/v1/stock/findAll?pageNum='+pageNum);
    }
    this.findPage=function (pageNum,pageSize) {
        return $http.get('api/v1/stock/findAll?pageNum='+pageNum+"&pageSize="+pageSize)
    }
    this.search=function (keywords) {
        return $http.get("../scenery/search?keywords="+keywords)
    }
});